eccentricdevotion/TARDIS
# Migrate gh-pages to use modern Jekyll site to serve files (via markdown)

After about a few hours of work, I've made this very large commit.

## What does it do?

It converts (most of) the old .html files into .md files. Making it easier and friendlier to code and work with!

## How does it work?

GitHub pages has been using Jekyll for a while. Now, instead of hard-coding every single HTML page, there's many tools to help such as layouts and includes, code you only need to write once and wherever you include it, it's already there. Helps prevent something like [this](https://github.com/eccentricdevotion/TARDIS/commit/d5ff5ba9c63f121ff4f6c8f7f775808871126417). Now you can just edit the _includs/header.html and it's now propagated to all the pages!

Each file is written in markdown, much like the READMEs are. This makes it much easier to write, as Jekyll handles converting it to HTML and hosting it.

Don't worry! You can still include HTML in all of the markdown files like usual, they'll be rendered properly.

I didn't convert every file, some couldn't make it through the converter. If this gets through I'll go through and fix those. This is mainly for me (or anyone else) who has a harder time reading and coding in HTML and much prefers simpler Markdown formatting to do code! I'll definitely go through and update some pages.

## Making new pages

Instead of pasting the HTML skeleton on every page, now all you need to do is add this to the top of the file.__md__ 

```yaml
---
layout: default
title: The title of the page
---
```

It will fill out all the HTML from the [defaults.html](https://github.com/Chew/TARDIS/blob/gh-pages/_layouts/default.html) file.

## Improvements over just using HTML

* As mentioned, you don't need to change every file to modify the head anymore
* Easier to read and write Markdown (instead of everything being in HTML)
* Pages can be `[url]/page` without needing the `.html`!

## The config.yml file

I've made a new file called `config.yml` this configures the note on the blue part of the top.
![image](https://user-images.githubusercontent.com/8278263/77827331-f8f5ce80-70e2-11ea-81e8-6306c86735d5.png)
In case you wanna change Spiget to Spigot, just change it in the config.yml and it'll automatically go everywhere!

## Testing Wiki changes locally

This may be a bit more tricky, since you now have to run Jekyll to test changes. Luckily, it's a breeze for us mac/linux users!

1) Open the Terminal and `cd` into wiki pages directory.
2) Run `bundle install`, this installs the gems needed to run.
3) Run `bundle exec jekyll serve`, this runs the jekyll server.
4) Navigate to `http://localhost:4000`, you can see all the changes here!

## Example Site

This is available on my fork found here: https://chew.pro/TARDIS. If merged, your TARDIS page will look *exactly* like this. Code also here: https://github.com/Chew/TARDIS/tree/gh-pages

Let me know what you think of it, it may look massive, but only because every single line was converted to markdown, there's hardly any HTML left :p
